src/Microsoft.Health.Dicom.Core/Features/Model/InstanceIdentifier.cs

```diff
@@ -17,34 +17,12 @@ public class InstanceIdentifier
         string studyInstanceUid,
         string seriesInstanceUid,
         string sopInstanceUid,
-        PartitionEntry partitionEntry)
-        : this(
-            studyInstanceUid,
-            seriesInstanceUid,
-            sopInstanceUid,
-            partitionEntry?.PartitionKey ?? throw new ArgumentNullException(nameof(partitionEntry)),
-            partitionEntry?.PartitionName ?? throw new ArgumentNullException(nameof(partitionEntry)))
+        Partition partition)
     {
-        EnsureArg.IsNotNull(partitionEntry, nameof(partitionEntry));
-    }
-
-    public InstanceIdentifier(
-        string studyInstanceUid,
-        string seriesInstanceUid,
-        string sopInstanceUid,
-        int partitionKey = DefaultPartition.Key,
-        string partitionName = DefaultPartition.Name)
-    {
-        EnsureArg.IsNotNullOrWhiteSpace(studyInstanceUid, nameof(studyInstanceUid));
-        EnsureArg.IsNotNullOrWhiteSpace(seriesInstanceUid, nameof(seriesInstanceUid));
-        EnsureArg.IsNotNullOrWhiteSpace(sopInstanceUid, nameof(sopInstanceUid));
-
-        StudyInstanceUid = studyInstanceUid;
-        SeriesInstanceUid = seriesInstanceUid;
-        SopInstanceUid = sopInstanceUid;
-        PartitionKey = partitionKey;
-        PartitionName = partitionName;
-        PartitionEntry = new PartitionEntry(PartitionKey, PartitionName);
+        Partition = EnsureArg.IsNotNull(partition, nameof(partition));
```

Comment:
Checking for null partition is correct, but are we ensuring that the key is not default?

Reply:
Because there is some inconsistency in the model, you check for null for name but for key you dont check if it greater than 0